### PR TITLE
Add a locked version of the sprite editor for custom objects

### PR DESCRIPTION
**Pull request by D8H**
http://gdevelop-storybook.s3-website-us-east-1.amazonaws.com/ebo-sprite-configuration/latest/index.html?path=/story/objecteditor-spriteeditor--animation-locked

image

### CODE REVIEW

**Comment by D8H** [comment]
> Is it normal that it looks like I can:
> 
>     * Add a sprite to an animation
> 
>     * Change the time between frames of an animation
> 
>     * Select a frame in an animation to delete it
> 
>     * Toggle the loop attribute

Yes, users should be able to define the animation content, but not change the animation list order and names.

File: newIDE/app/src/ObjectEditor/Editors/SpriteEditor/index.js [hunk 1]
```diff
@@ -75,28 +77,37 @@ class Animation extends React.Component<AnimationProps, void> {
...
           </Column>
-          <IconButton size="small" onClick={onRemove}>
-            <Delete />
-          </IconButton>
-        </MiniToolbar>
+        )}
+        {!isAnimationListLocked && (
+          <MiniToolbar noPadding>
+            <DragHandle />
+            <MiniToolbarText>{<Trans>Animation #{id}</Trans>}</MiniToolbarText>
```
Comment (4ian) [inline newIDE/app/src/ObjectEditor/Editors/SpriteEditor/index.js:95]: Thanks!

File: newIDE/app/src/ObjectEditor/Editors/SpriteEditor/index.js [hunk 2]
```diff
@@ -75,28 +77,37 @@ class Animation extends React.Component<AnimationProps, void> {
...
           <Column expand>
-            <SemiControlledTextField
-              commitOnBlur
-              margin="none"
-              value={animation.getName()}
-              translatableHintText={t`Optional animation name`}
-              onChange={text => onChangeName(text)}
-              fullWidth
-            />
+            <Spacer />
```
Comment (4ian) [inline newIDE/app/src/ObjectEditor/Editors/SpriteEditor/index.js:88]: Could we do without this spacer?

Comment (4ian) [reply newIDE/app/src/ObjectEditor/Editors/SpriteEditor/index.js:88]: (Column already has some spacing on the left/right)

File: newIDE/app/src/ObjectEditor/ObjectsEditorService.js [hunk 3]
```diff
@@ -35,6 +35,32 @@ const ObjectsEditorService = {
...
+      return this.getCustomObjectPropertiesEditor({
+        helpPagePath: objectMetadata.getHelpPath(),
+      });
+    }
+    console.warn(
+      `Object with type ${objectType} has no editor configuration registered. Please use registerEditorConfiguration to register your editor.`
+    );
+    return this.getDefaultObjectJsImplementationPropertiesEditor({
+      helpPagePath: '',
+    });
```
Comment (4ian) [inline newIDE/app/src/ObjectEditor/ObjectsEditorService.js:62]: Could all of this be a fallback call to `getEditorConfiguration`?

**Review by 4ian — APPROVED** [review]
Overall lgtm!

File: newIDE/app/src/ObjectEditor/Editors/SpriteEditor/index.js (hunk 2 above)
Comment (D8H) [reply newIDE/app/src/ObjectEditor/Editors/SpriteEditor/index.js:88]: It's for the top, the title was right under the previous animation sprite.

Comment (4ian) [reply newIDE/app/src/ObjectEditor/Editors/SpriteEditor/index.js:88]: Ok, check if a Line can do, otherwise it's ok

Comment (D8H) [reply newIDE/app/src/ObjectEditor/Editors/SpriteEditor/index.js]: A Line works

**Review by 4ian — APPROVED** [review]

**Pull request merged**